pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -823,6 +824,14 @@ func (vs *vSphereVMProvider) updateVirtualMachine(
 		reconcileErr = getReconcileErr("power state", reconcileErr, err)
 	}
 
+	// create snapshots of VM if necessary
```

Comment:
```suggestion
	// Reconcile the current snapshot of this VM.
```

Because this can essentially be:
- take a snapshot
- don't do anything (just patch status)
- revert